group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -10921,6 +10822,1048 @@ public void testLastConsumerProtocolMemberRebalanceTimeoutInConsumerGroup() {
+                    .build())
+                .withConsumerGroup(new ConsumerGroupBuilder(groupId, 11)
+                    .withSubscriptionMetadata(new HashMap<String, TopicMetadata>() {
+                        {
+                            put(fooTopicName, new TopicMetadata(fooTopicId, fooTopicName, 2, mkMapOfPartitionRacks(2)));
+                            put(barTopicName, new TopicMetadata(barTopicId, barTopicName, 1, mkMapOfPartitionRacks(1)));
+                        }
+                    })
+                    .withMember(new ConsumerGroupMember.Builder(memberId1)
+                        .setState(MemberState.UNREVOKED_PARTITIONS)
+                        .setMemberEpoch(10)
+                        .setPreviousMemberEpoch(10)
+                        .setRebalanceTimeoutMs(500)
+                        .setClientId("client")
+                        .setClientHost("localhost/127.0.0.1")
+                        .setSubscribedTopicNames(Arrays.asList(fooTopicName, barTopicName))
+                        .setSupportedClassicProtocols(GroupMetadataManagerTestContext.toRangeProtocol(
+                            Arrays.asList(fooTopicName, barTopicName),
+                            Arrays.asList(new TopicPartition(fooTopicName, 0), new TopicPartition(fooTopicName, 1)),
+                            version))
+                        .setPartitionsPendingRevocation(mkAssignment(
+                            mkTopicAssignment(fooTopicId, 1)))
+                        .build())
+                    .withMember(new ConsumerGroupMember.Builder(memberId2)
+                        .setState(MemberState.STABLE)
+                        .setMemberEpoch(10)
+                        .setPreviousMemberEpoch(10)
+                        .setRebalanceTimeoutMs(500)
+                        .setClientId("client")
+                        .setClientHost("localhost/127.0.0.1")
+                        .setSubscribedTopicNames(Arrays.asList(fooTopicName, barTopicName))
+                        .setPartitionsPendingRevocation(mkAssignment(
+                            mkTopicAssignment(barTopicId, 0)))
+                        .build())
+                    .withAssignment(memberId1, mkAssignment(
+                        mkTopicAssignment(fooTopicId, 0),
+                        mkTopicAssignment(barTopicId, 0)))
+                    .withAssignment(memberId2, mkAssignment(
+                        mkTopicAssignment(fooTopicId, 1)))
+                    .withAssignmentEpoch(11))
+                .build();
+            ConsumerGroup group = context.groupMetadataManager.consumerGroup(groupId);
+            group.setMetadataRefreshDeadline(Long.MAX_VALUE, 11);
+
+            JoinGroupRequestData request = new GroupMetadataManagerTestContext.JoinGroupRequestBuilder()
+                .withGroupId(groupId)
+                .withMemberId(memberId1)
+                .withProtocols(GroupMetadataManagerTestContext.toRangeProtocol(
```

Comment:
nit: The Range assignor never reports owned partitions because it always revoked all of them before rejoining. We should perhaps use `toConsumerProtocol`.